src/App.js

```diff
@@ -3,35 +3,42 @@ import { Switch, Route, NavLink } from "react-router-dom";
 import Age from "./Age";
 import Nation from "./Nation";
 import Gender from "./Gender";
+import InputBar from "./InputBar";
+import { useState } from "react";
 
 function App() {
+  const [currentName, setCurrentName] = useState("");
   return (
     <div className="App">
-      <header class="app__header">
-        <Switch>
-          <Route path="/age">
-            <Age />
-          </Route>
-          <Route path="/nation">
-            <Nation />
-          </Route>
-          <Route path="/gender">
-            <Gender />
-          </Route>
-        </Switch>
-      </header>
-      <main class="app__main">
-        <input class="app__input" text="text" placeholder="Type name..." />
-        <p>
-          <button>Submit</button>
-        </p>
+      <header className="app__header">Guess...</header>
+      <main className="app__main">
+        <InputBar onNameChange={(name) => setCurrentName(name)} />
+        <div className="app__content--results">
+          <Switch>
+            <Route path="/age">
+              <Age name={currentName} />
+            </Route>
+            <Route path="/nation">
+              <Nation name={currentName} />
+            </Route>
+            <Route path="/gender">
+              <Gender name={currentName} />
+            </Route>
+          </Switch>
+        </div>
```

Comment:
Cool!
You solved it in a different way and you put the state in the App.js so you can share it with the different pages. That's good thinking! 🥳